### PR TITLE
Refactors reviver code

### LINKED ISSUE #80371: Reviver implant can revive you despite being in an invalid state (decapitated)
## Issue Summary

In certain circumstances, the reviver implant can revive you from death in an invalid state, such as decapitated. 

## Reproduction:

At a glance, the code for the reviver implant does the following: 

- Try to heal the mob.
- If they are dead, check if they can be defibbed. 
- If they can be defibbed, defib them next life tick.

Essentially: If the mob is in a valid state THIS LIFE TICK, revive them NEXT LIFE TICK regardless. 

https://github.com/tgstation/tgstation/blob/097c140dea9680ebb73291998ebfef0926c70b75/code/modules/surgery/organs/internal/cyberimp/augments_chest.dm#L86-L96

But what happens if, between ticks, you become unable to be defibbed? There is no guard against this, so it revives you regardless.

### PR DESCRIPTION
**Pull request by Diamond-74**
## About The Pull Request

Cleans up reviver code to overall make it simpler and properly implement the cooldown that it said it had in the code but never really had. Additionally, stops the reviver trying to revive someone that isn't in a revivable state, which fixes https://github.com/tgstation/tgstation/issues/80371. Wasn't sure how long to make the delay but I figured just a 5 second delay after the first heal proc should be fine. About the same as what the original cooldown was from being paired to the heal proc's second iteration.
## Why It's Good For The Game

Changes variables and checks used by the reviver to make sure it functions properly with the cooldown that seemed to be intended in the code. Additionally, prevents the reviver from infinitely defibbing a non-revivable corpse.
## Changelog
:cl:
fix: Reviver no longer attempts to revive impossible to defib mobs.
refactor: Cleaned up unnecessary variables and re-arraigned code to have it perform altogether in one tick. Additionally added a proper cooldown to revivers.
/:cl:

### CODE REVIEW

**Comment by AdipemDragon** [comment]
If you want it to link, you need "Fixes #80371" somewhere in the body, not in the title

**Comment by Diamond-74** [comment]
Alright, thanks

**Comment by SyncIt21** [comment]
You didn't do it correctly. Remove `Fixes #80371` from your PR title and put it in your PR body. 

Replace this

> resolving #80371 

With this

> Fixes #80371

File: code/modules/surgery/organs/internal/cyberimp/augments_chest.dm [hunk 1]
```diff
@@ -119,16 +112,21 @@
...
-
-	owner.visible_message(span_warning("[owner]'s body convulses a bit."))
-	playsound(owner, SFX_BODYFALL, 50, TRUE)
-	playsound(owner, 'sound/machines/defib_zap.ogg', 75, TRUE, -1)
-	owner.revive()
-	owner.emote("gasp")
-	owner.set_jitter_if_lower(200 SECONDS)
-	SEND_SIGNAL(owner, COMSIG_LIVING_MINOR_SHOCK)
-	log_game("[owner] been revived by [src]")
+	if(COOLDOWN_FINISHED(src, defib_cooldown) && owner.stat == DEAD && owner.can_defib() == DEFIB_POSSIBLE)
```
Comment (Ghommie) [inline code/modules/surgery/organs/internal/cyberimp/augments_chest.dm:115]: Do an early return please.

**Review by Ghommie — CHANGES_REQUESTED** [review]

File: code/modules/surgery/organs/internal/cyberimp/augments_chest.dm (hunk 1 above)
Comment (Diamond-74) [reply code/modules/surgery/organs/internal/cyberimp/augments_chest.dm:115]: Alright.

Comment (Diamond-74) [reply code/modules/surgery/organs/internal/cyberimp/augments_chest.dm]: I believe the change I made is what you wanted, if not let me know..

**Pull request merged**